core/shared/src/main/scala/monocle/function/Index.scala

```diff
@@ -27,6 +27,11 @@ trait IndexFunctions {
 }
 
 object Index extends IndexFunctions{
+
+  def apply[S, I, A](getOption: I => S => Option[A])(set: I => A => S => S): Index[S, I, A] = new Index[S, I, A] {
```

Comment:
why did you inline the `Optional` definition? it makes the API inconsistent and potentially you can create Optional via other ways

Reply:
I did it like that to be consistent with `At.apply`. But I can change it to `I => Optional[S, A]` or maybe have both?

Reply:
interesting, I completely forgot that we had `At.apply`. It is weird because it doesn't even look that we use it. 

Would you mind to add `I => Lens[S, A]` for `At` and use `I => Optional[S, A]` for `Index`? Maybe we can deprecate `At.apply` or add another apply to `Index` in another PR

Reply:
Done.
Apparently `At.apply` was used in `refined.BitsInstances`. I migrated the use to new `apply` method.